contracts/src/token/erc20/extensions/flash_mint.rs

```diff
@@ -0,0 +1,318 @@
+            /// the `keccak256` hash of "ERC3156FlashBorrower.onFlashLoan".
+            ///
+            /// # Arguments
+            ///
+            /// * `initiator` - The address which initiated the flash loan.
+            /// * `token` - The address of the token being loaned (loan currency).
+            /// * `amount` - The amount of tokens lent in the flash loan.
+            /// * `fee` - The additional fee to repay with the flash loan amount.
+            /// * `data` - Arbitrary data structure, intended to contain user-defined parameters.
+            #[allow(missing_docs)]
+            function onFlashLoan(
+                address initiator,
+                address token,
+                uint256 amount,
+                uint256 fee,
+                bytes calldata data
+            ) external returns (bytes32);
+        }
+    }
+}
+
+/// State of the [`Erc20FlashMint`] Contract.
+#[storage]
+pub struct Erc20FlashMint {
+    /// Fee applied when doing flash loans.
+    pub flash_fee_amount: StorageU256,
+    /// Receiver address of the flash fee.
+    pub flash_fee_receiver_address: StorageAddress,
+}
+
+/// Extension of [`Erc20`] that allows token holders to destroy both
+/// their own tokens and those that they have an allowance for,
+/// in a way that can be recognized off-chain (via event analysis).
+pub trait IERC3156FlashLender {
+    /// The error type associated to this ERC-20 Burnable trait implementation.
+    type Error: Into<alloc::vec::Vec<u8>>;
+
+    /// Returns the maximum amount of tokens that can be borrowed from this
+    /// contract in a flash loan.
+    ///
+    /// For tokens that are not supported, this function returns `U256::MIN`.
+    ///
+    /// * `token` - The address of the ERC-20 token that will be loaned.
+    /// * `erc20` - Read access to a contract providing [`IErc20``] interface.
+    fn max_flash_loan(&self, token: Address, erc20: &Erc20) -> U256;
+
+    /// Calculates the fee for a flash loan.
+    ///
+    /// The fee is a fixed percentage of the borrowed amount.
+    ///
+    /// If the token is not supported, the function returns an
+    /// `UnsupportedToken` error.
+    ///
+    /// * `token` - The address of the ERC-20 token that will be loaned.
+    /// * `amount` - The amount of tokens that will be loaned.
+    fn flash_fee(
+        &self,
+        token: Address,
+        amount: U256,
+    ) -> Result<U256, Self::Error>;
+
+    /// Executes a flash loan.
+    ///
+    /// This function is part of the ERC-3156 (Flash Loans) standard.
+    ///
+    /// * `receiver` - The contract that will receive the flash loan.
+    /// * `token` - The ERC-20 token that will be loaned.
+    /// * `amount` - The amount of tokens that will be loaned.
+    /// * `data` - Arbitrary data that can be passed to the receiver contract.
+    ///
+    /// The function must return `true` if the flash loan was successful,
+    /// and revert otherwise.
+    fn flash_loan(
```

Comment:
Improve Rust docs:
- [ ] Add `# Arguments` section;
- [ ] Mention `&self` in arguments;
- [ ] Missing `# Errors` section;
- [ ] Missing `# Panics` / `# Events` section if applicable;